pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -133,6 +134,25 @@ func (c TargetConfigController) sync(ctx context.Context, syncContext factory.Sy
 	}
 
 	requiredCM, err := c.applyClusterCSIDriverChange(infra, cfg, clusterCSIDriver)
+	cmString := string(c.manifest)
+
+	virtualCenterIPs := sets.StringKeySet(cfg.VirtualCenter)
+
+	if len(virtualCenterIPs) != 1 {
+		return fmt.Errorf("cloud config %s/%s must define a single VirtualCenter", cloudConfigNamespace, cloudConfig.Name)
+	}
+
+	virtualCenterIP := virtualCenterIPs.List()[0]
+
+	for pattern, value := range map[string]string{
+		"${CLUSTER_ID}":  infra.Status.InfrastructureName,
+		"${VCENTER}":     virtualCenterIP,
+		"${DATACENTERS}": cfg.VirtualCenter[virtualCenterIP].Datacenters,
+	} {
+		cmString = strings.ReplaceAll(cmString, pattern, value)
+	}
+
+	requiredCM := resourceread.ReadConfigMapV1OrDie([]byte(cmString))
```

Comment:
This is making change in outdated code. Can you check code again and update?

Reply:
hi @gnufied odd, i just rebased

Reply:
i'll check again